### PR TITLE
Support options

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
 import { Source } from 'callbag'
 
-export default function fromEvent(node: EventTarget, name: string): Source<Event>;
+export default function fromEvent<T extends Event = Event>(
```
Comment (Andarist) [inline types.d.ts:3]: please review this change extra carefully - im not a TS expert, beside adjusting the types to account for new argument I've also made the whole thing generic, which makes sense, right? MAYBE this even could be improved further by using conditional types, but this is over my head right now

Comment (staltz) [reply types.d.ts:3]: Looks safe. The best would be to implement in TypeScript, not JavaScript, but I believe this will work

**Comment by staltz** [comment]
Could you rebase with master?

**Comment by Andarist** [comment]
Rebased

**Comment by staltz** [comment]
Thanks!

**Pull request merged**